sklearn/decomposition/_kernel_pca.py

```diff
@@ -317,7 +317,7 @@ def _fit_transform(self, K):
         else:
             if self.n_components < 1:
                 raise ValueError(
-                    f"`n_components` should be >= 1, got: {self.n_component}"
+                    f"`n_components` should be >= 1, got: {self.n_components}"
                 )
```

Comment:
May we add a test in `sklearn/decomposition/tests/test_kernel_pca.py` testing that the error message is raised correctly for `self.n_components==0`?

Also since we are fixing this big, we can use `check_scalar` here:

```python
# import at the top of file
import numbers
from ..utils.validation import check_scalar

if self.n_components is None:
    n_components = K.shape[0]  # use all dimensions
else:
    check_scalar(self.n_components, numbers.Integral, min_val=1)
    n_components = min(K.shape[0], self.n_components)
```

Reply:
Yes of course. This is my first Pull Request, so do I have to do something (i.e. add the proposed changes to the pull request)? :)

Reply:
Yea, you may update this PR by making the changes to your branch.

Reply:
@DataDave1337 Are you still interested in working on this PR?